src/iexamine.cpp

```diff
@@ -974,23 +974,35 @@ void iexamine::slot_machine(player *p, map*, const tripoint&)
 void iexamine::safe(player *p, map *m, const tripoint &examp)
 {
     if (!p->has_amount("stethoscope", 1)) {
-        add_msg(m_info, _("You need a stethoscope for safecracking."));
-        return;
+        if (query_yn(_("The safe is locked. Input a random combination?"))) {
+            p->moves -= 100;
+            // one_in(30^3) chance of guessing
+            if (one_in(27000)) {
+                p->add_msg_if_player(m_good, _("The safe opens!"));
+                m->furn_set(examp, f_safe_o);
+                return;
+            } else {
+                p->add_msg_if_player(m_info, _("The safe remains locked."));
+                return;
+            }
+        } else {
+            p->add_msg_if_player(m_info, _("Never mind."));
+            return;
+        }
     }
 
     if (query_yn(_("Attempt to crack the safe?"))) {
-        bool success = true;
         if (p->is_deaf()) {
             add_msg(m_info, _("You can't crack a safe while deaf!"));
             return;
         }
+         // 150 minutes +/- 20 minutes per mechanics point away from 3 +/- 10 minutes per
+        // perception point away from 8; capped at 30 minutes minimum.
+        int moves = std::max(150000 + (p->skillLevel("mechanics") - 3) * -20000 +
+                             (p->get_per() - 8) * -10000, 30000);
```

Comment:
This formula might be easier to read if you use the `MINUTES` macro. It converts minutes to turns (not moves, you have to multiply the result with 100). It's a bit more obvious what you mean with `moves = MINUTES(18) * 100`, through the `100` is still rather obscure;

Ideally, there should be a `MOVES` or `TURNS_TO_MOVES` inline function that does the `*100` - or even better a class that represents turn counts and one that represents move counts and implicit conversion between. So one can write `Moves moves = MINUTES(10);` and it automatically becomes `Moves = (10 * 6) * 100;` - `MINUTES` returning a `TurnCount` (or similar) object (including the `*6`) that is automatically converted to a `Moves` object (including the `*100`).

Well, it's a nice dream.